### PR TITLE
feat: addressbook overlay

### PR DESCRIPTION
**Pull request by simboonlong**
use these files to test:
- test/fixture/maersk-bill-of-lading.tt (upload tt)
- test/fixture/local-addressbook.csv (upload addressbook)

features:
- add overlay default
- add overlay addressbook
- shift csv import to addressbook overlay
- search input to filter addressbook by names or addresses
- add react-transitions

### CODE REVIEW

File: src/components/MultiTabs.js [hunk 1]
```diff
@@ -1,49 +1,77 @@
...
-                    selectTemplateTab(idx);
+                    setOverlayVisible(!isOverlayVisible);
                   }}
-                  role="tab"
-                  aria-controls="home"
-                  aria-selected="true"
                 >
-                  {t.label}
-                </a>
+                  Upload Address Book
```
Comment (yehjxraymond) [inline src/components/MultiTabs.js]:
```suggestion
                  Address Book
```

It makes sense to call it address book now that it doesnt open the dialog immediately.

**Comment by yehjxraymond** [comment]
image

File: package.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
     "dev-cli": "node -r esm ./scripts/cli.js",
     "lint": "eslint ./src --ext .js,.ts,.tsx,.mdx --max-warnings 0",
     "lint:fix": "npm run lint -- --fix",
+    "int": "testcafe chrome",
```
Comment (yehjxraymond) [inline package.json]: Why is this needed?

File: src/components/MultiTabs.js [hunk 3]
```diff
@@ -1,49 +1,77 @@
...
-                <a
-                  className={`${styles.tab}
-                    ${idx === activeTab ? styles.active : ""}`}
-                  id={t.id}
+const MultiTabs = ({ activeTab, templates, selectTemplateTab }) => {
+  const [isOverlayVisible, setOverlayVisible] = useState(false);
+  const { addressBook } = useAddressBook();
+
+  return (
+    <>
```
Comment (yehjxraymond) [inline src/components/MultiTabs.js]: Is there a need for fragments here?

File: src/components/MultiTabs.js [hunk 4]
```diff
@@ -1,49 +1,77 @@
...
+      <div id={styles["header-ui"]} className="pt-3 pt-md-0">
+        <div className={`${styles["header-container"]}`}>
+          <ul id="template-tabs-list" className="nav nav-tabs row no-gutters align-items-center">
+            <li className="nav-item col-auto col-md-auto ml-md-auto order-md-2">
+              <a href=" " className="my-auto ml-auto">
+                <ButtonBordered bg="tertiary">View another</ButtonBordered>
+              </a>
+            </li>
+            <FeatureFlag name="ADDRESS_BOOK_UPLOAD">
```
Comment (yehjxraymond) [inline src/components/MultiTabs.js]: Can remove the feature flag now, the whole `ADDRESS_BOOK_UPLOAD` flag from all the code since we want this permanently on from now.

File: src/components/MultiTabs.js [hunk 5]
```diff
@@ -1,49 +1,77 @@
...
                 >
-                  {t.label}
-                </a>
+                  Upload Address Book
+                </ButtonBordered>
+                <OverlayAddressBook
+                  id="overlay-addressbook"
+                  title="Address Book"
+                  isOverlayVisible={isOverlayVisible}
+                  onClick={() => {
```
Comment (yehjxraymond) [inline src/components/MultiTabs.js]:
```suggestion
                  handleCloseOverlay={() => {
```

File: src/components/MultiTabs.js [hunk 6]
```diff
@@ -1,49 +1,77 @@
...
-                  aria-selected="true"
                 >
-                  {t.label}
-                </a>
+                  Upload Address Book
+                </ButtonBordered>
+                <OverlayAddressBook
+                  id="overlay-addressbook"
+                  title="Address Book"
+                  isOverlayVisible={isOverlayVisible}
```
Comment (yehjxraymond) [inline src/components/MultiTabs.js]: Do we need to pass this in or we can use this outside? 

```
{isOverlayVisible ? <MyComponentHere />}
```

File: src/components/MultiTabs.js [hunk 7]
```diff
@@ -1,49 +1,77 @@
...
+                  onClick={() => {
+                    setOverlayVisible(!isOverlayVisible);
+                  }}
+                >
+                  <OverlayAddressBookContent addressBook={addressBook} />
```
Comment (yehjxraymond) [inline src/components/MultiTabs.js]: Can we put this content within the `OverlayAddressBook` component and pass in `addressBook` into the `OverlayAddressBook` component?

File: src/components/UI/Overlay/Overlay.tsx [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+      {children}
+    </Overlay>
+  );
+};
+
+interface OverlayAddressBookContentProps {
+  addressBook: AddressBook;
+}
+
+export const OverlayAddressBookContent = ({ addressBook }: OverlayAddressBookContentProps) => {
```
Comment (yehjxraymond) [inline src/components/UI/Overlay/Overlay.tsx]: This component doesnt seemed to belong to the folder `UI/Overlay`

File: src/components/UI/Overlay/Overlay.tsx [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+              <div className="col-12">{children}</div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export const OverlayAddressBook = ({ children, ...props }: OverlayProps) => {
```
Comment (yehjxraymond) [inline src/components/UI/Overlay/Overlay.tsx]: This component doesnt seemed to belong to the folder `UI/Overlay`

File: src/components/UI/Overlay/Overlay.tsx [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+  return (
+    <Overlay className={`${css.addressbook}`} {...props}>
+      {children}
```
Comment (yehjxraymond) [inline src/components/UI/Overlay/Overlay.tsx]: Can put the `OverlayAddressBookContent` in here

File: src/components/UI/Overlay/Overlay.tsx [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+  addressBook: AddressBook;
+}
+
+export const OverlayAddressBookContent = ({ addressBook }: OverlayAddressBookContentProps) => {
+  const [searchName, setSearchName] = useState("");
+  const [filteredAddressBook, setFilteredAddressBook] = useState(addressBook);
+
+  useEffect(() => {
+    if (searchName.length > 0) {
+      const filtered = pickBy(addressBook, value => {
```
Comment (yehjxraymond) [inline src/components/UI/Overlay/Overlay.tsx]:
```suggestion
      const filtered = pickBy(addressBook, addressBookEntry => {
```

File: src/components/UI/Overlay/Overlay.tsx [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+        const searchTerm = searchName.toLowerCase();
+        const name = value.toLowerCase();
+        if (name.includes(searchTerm)) {
```
Comment (yehjxraymond) [inline src/components/UI/Overlay/Overlay.tsx]: do we want to search the key (filter by address) too?

File: src/components/UI/Overlay/Overlay.tsx [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+
+interface OverlayAddressBookContentProps {
+  addressBook: AddressBook;
+}
+
+export const OverlayAddressBookContent = ({ addressBook }: OverlayAddressBookContentProps) => {
+  const [searchName, setSearchName] = useState("");
+  const [filteredAddressBook, setFilteredAddressBook] = useState(addressBook);
+
+  useEffect(() => {
```
Comment (yehjxraymond) [inline src/components/UI/Overlay/Overlay.tsx]: Dont need useEffect i think, can just let the whole thing rerender.

File: src/components/UI/Overlay/Overlay.tsx [hunk 14]
```diff
@@ -0,0 +1,149 @@
...
+    </Overlay>
+  );
+};
+
+interface OverlayAddressBookContentProps {
+  addressBook: AddressBook;
+}
+
+export const OverlayAddressBookContent = ({ addressBook }: OverlayAddressBookContentProps) => {
+  const [searchName, setSearchName] = useState("");
```
Comment (yehjxraymond) [inline src/components/UI/Overlay/Overlay.tsx]:
```suggestion
  const [searchTerm, setSearchTerm] = useState("");
```

File: package.json (hunk 2 above)
Comment (simboonlong) [reply package.json]: i cant find on how to run a specific single test in testcafe, so now what i do is `npm run int <path-to-test>`, the path i copy paste relative path via vsc.

File: src/components/MultiTabs.js (hunk 6 above)
Comment (simboonlong) [reply src/components/MultiTabs.js]: mmm, think i need to pass it in cos i want to toggle a css class `is-visible`, which in turn controls the fading + scale animation of the overlay.

File: src/components/UI/Overlay/Overlay.tsx (hunk 12 above)
Comment (simboonlong) [reply src/components/UI/Overlay/Overlay.tsx]: ok sure

File: package.json (hunk 2 above)
Comment (yehjxraymond) [reply package.json]: My concern is "int" doesnt say what you just did. Might want to have a better name for it and maybe add into the readme what it does?

File: src/components/UI/Overlay/Overlay.tsx [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+import css from "./Overlay.module.scss";
+import { SvgIcon, SvgIconX, SvgIconSearch } from "../../UI/SvgIcon";
+import { CsvUploadButton } from "../../AddressBook/CsvUploadButton";
+import { AddressBook } from "../../../common/hooks/useAddressBook";
+import { makeEtherscanAddressURL } from "../../../utils";
+import { isEmpty, pickBy } from "lodash";
+
+interface OverlayProps {
+  id?: string;
+  isOverlayVisible: boolean;
```
Comment (yehjxraymond) [inline src/components/UI/Overlay/Overlay.tsx]: Usually visibility is controlled by the parent component, when it's not visible it wont even render. Something like:

```
const OverlayParent = () => {
...
  return(
    <Something /> 
    {showOverlay && <Overlay />}
    <SomethingElse /> 
  )
}
```

Is there a reason you chose to let the child decide to show or now?

Comment (simboonlong) [reply src/components/UI/Overlay/Overlay.tsx]: the above method would be a show/hide treatment with no transitions.

i've implemented CSSTransition from react-transitions instead, to better manage transitions when components mount and unmount, while preserving it to be not rendered when it's not visible

**Review by yehjxraymond — APPROVED** [review]
LGTM, nice sexy animation too (=
